prog/vm/github_runner.rb

```diff
@@ -62,6 +62,31 @@ def pick_vm
     vm_st.subject
   end
 
+  def find_closest_docker_mirror_server
+    compatible_servers = DockerRegistryMirrorServer.all.select { |server| server.vm.arch == vm.arch }
```

Comment:
nit: N+1 queries. Another option you can move filter to db side
```suggestion
    compatible_servers = DockerRegistryMirrorServer.eager(:vm).all.select { |server| server.vm.arch == vm.arch }
```